src/ECDSARegistryCoordinator.sol

```diff
@@ -0,0 +1,867 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import {OwnableUpgradeable} from "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import {Initializable} from "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+import {EIP1271SignatureUtils} from "eigenlayer-contracts/src/contracts/libraries/EIP1271SignatureUtils.sol";
+import {IPauserRegistry} from "eigenlayer-contracts/src/contracts/interfaces/IPauserRegistry.sol";
+import {Pausable} from "eigenlayer-contracts/src/contracts/permissions/Pausable.sol";
+
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {ISocketUpdater} from "./interfaces/ISocketUpdater.sol";
+import {IStakeRegistry} from "./interfaces/IStakeRegistry.sol";
+import {IServiceManager} from "./interfaces/IServiceManager.sol";
+
+import {BitmapUtils} from "./libraries/BitmapUtils.sol";
+
+/**
+ * @title A `RegistryCoordinator` that has three registries:
+ *      1) a `StakeRegistry` that keeps track of operators' stakes
+ *      2) a `BLSApkRegistry` that keeps track of operators' BLS public keys and aggregate BLS public keys for each quorum
+ *      3) an `IndexRegistry` that keeps track of an ordered list of operators for each quorum
+ * 
+ * @author Layr Labs, Inc.
+ */
+contract ECDSARegistryCoordinator is 
+    EIP712, 
+    Initializable, 
+    Pausable,
+    OwnableUpgradeable,
+    ISocketUpdater, 
+    ISignatureUtils
+{
+    using BitmapUtils for *;
+
+    // EVENTS
+
+    /// Emits when an operator is registered
+    event OperatorRegistered(address indexed operator, bytes32 indexed operatorId);
+
+    /// Emits when an operator is deregistered
+    event OperatorDeregistered(address indexed operator, bytes32 indexed operatorId);
+
+    event OperatorSetParamsUpdated(uint8 indexed quorumNumber, OperatorSetParam operatorSetParams);
+
+    event ChurnApproverUpdated(address prevChurnApprover, address newChurnApprover);
+
+    event EjectorUpdated(address prevEjector, address newEjector);
+
+    /// @notice emitted when all the operators for a quorum are updated at once
+    event QuorumBlockNumberUpdated(uint8 indexed quorumNumber, uint256 blocknumber);
+
+    // DATA STRUCTURES
+    enum OperatorStatus
+    {
+        // default is NEVER_REGISTERED
+        NEVER_REGISTERED,
+        REGISTERED,
+        DEREGISTERED
+    }
+
+    // STRUCTS
+
+    /**
+     * @notice Data structure for storing info on operators
+     */
+    struct OperatorInfo {
+        // the id of the operator, which is likely the keccak256 hash of the operator's public key if using BLSRegistry
+        bytes32 operatorId;
```

Comment:
Still a bit confused as to whether this should be stored as a bytes32 or an address. 
```
which is likely the keccak256 hash of the operator's public key if using BLSRegistry
```
shows that this was copied from the BLSRegistry. Pretty convinced this should be address here right? (hash of ecdsa pubkey IS the ecdsa address by defn)